### ISSUE TITLE
Can't run madx instance. TypeError 

### ISSUE THREAD

**Issue by JohnMSmith** [hibtc/cpymad#110]
I have latest version of cpymad on Win10.
Python 3.9.12

Attempt to run mad leads to following output.

```
from cpymad.madx import Madx
madx = Madx()
```

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [31], in <cell line: 2>()
      1 from cpymad.madx import Madx
----> 2 madx = Madx()

File ~\Anaconda3\lib\site-packages\cpymad\madx.py:192, in Madx.__init__(self, libmadx, command_log, stdout, history, prompt, **Popen_args)
    190 if not libmadx.is_started():
    191     with self.reader:
--> 192         libmadx.start()
    193 # init instance variables:
    194 self.history = history

File ~\Anaconda3\lib\site-packages\cpymad\stream.py:57, in AsyncReader.__exit__(self, *exc_info)
     55 output_lines = self.result.get()
     56 if output_lines:
---> 57     self.callback(b''.join(output_lines))

File ~\Anaconda3\lib\site-packages\ipykernel\iostream.py:513, in OutStream.write(self, string)
    503 """Write to current stream after encoding if necessary
    504 
    505 Returns
   (...)
    509 
    510 """
    512 if not isinstance(string, str):
--> 513     raise TypeError(
    514         f"write() argument must be str, not {type(string)}"
    515     )
    517 if self.echo is not None:
    518     try:

TypeError: write() argument must be str, not <class 'bytes'>
```

How can I solve this?

**Comment by JohnMSmith** [comment]
cpymad was installed using
`pip install cpymad`

I've tried to build cpymad for python 3.9 using guide, but getting this while building cpymad

> call %gcc% -mdll -O -Wall ^
>     -I %MADXDIR%\include ^
>     -I %pythondir%\include ^
>     -c src/cpymad/libmadx.c ^
>     -o %tempdir%\libmadx.obj ^
>     -std=gnu99

> src/cpymad/libmadx.c:252:41: warning: division by zero [-Wdiv-by-zero]
>      enum { __pyx_check_sizeof_voidp = 1 / (int)(SIZEOF_VOID_P == sizeof(void*)) };
>                                          ^
> src/cpymad/libmadx.c:252:12: error: enumerator value for '__pyx_check_sizeof_voidp' is not an integer constant
>      enum { __pyx_check_sizeof_voidp = 1 / (int)(SIZEOF_VOID_P == sizeof(void*)) };

**Comment by JohnMSmith** [comment]
Was trying to install cpymad using conda:
```
conda skeleton pypi cpymad
conda build cpymad
conda install --use-local cpymad
```
but were unable to create a recipe for build. Got the following error:
> src/cpymad/libmadx.c(718): fatal error C1083: Cannot open include file: madX/mad_types_f.h: No such file or directory,
> error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.33.31629\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
> $PYTHONPATH = D:\TMP\tmpklprtgecconda_skeleton_cpymad-1.9.3.tar.gz\cpymad-1.9.3
> 
> Leaving build/test directories:
>   Work:
>  C:\Users\skamarokhama\Anaconda3\conda-bld\skeleton_1662533987044\work
>   Test:
>  C:\Users\skamarokhama\Anaconda3\conda-bld\skeleton_1662533987044\test_tmp
> Leaving build/test environments:
>   Test:
> source activate  C:\Users\skamarokhama\Anaconda3\conda-bld\skeleton_1662533987044\_test_env
>   Build:
> source activate  C:\Users\skamarokhama\Anaconda3\conda-bld\skeleton_1662533987044\_build_env
> 
> 
> Error: command failed: C:\Users\skamarokhama\Anaconda3\conda-bld\skeleton_1662533987044\_h_env\python.exe setup.py install

**Comment by coldfix** [comment]
Looks like jupyter stdout is set up as text stream for you. Might be a difference between windows and linux because I have never tried on windows. I have not tried it on windows so far. What's your jupyter version?

**Comment by JohnMSmith** [comment]
> What's your jupyter version?

Jupyter Notebook 6.4.8

**Comment by coldfix** [comment]
It seems jupyter has changed its behaviour. On linux the problem also exists (`sys.stdout.write` only accepts `str`), but only on python 3.7 and above. However on linux this doesn't even surface as TypeError because the stream is set up with a fileno that accepts bytes and writes to the terminal, i.e. `sys.stdout.write(str)` and `os.write(sys.stdout.fileno(), bytes)` do very different things.

I will release a bugfix shortly.

Thanks for reporting.

**Comment by coldfix** [comment]
Should be fixed in version v1.10. Can you confirm @JohnMSmith ?

**Comment by JohnMSmith** [comment]
Yes, it is fixed. Thanks!
